src/org/zaproxy/zap/extension/exportreport/ExportReportAPI.java

```
@@ -0,0 +1,184 @@
+
+    private static final String PREFIX = "exportreport";
+
+    private static final String ACTION_GENERATE = "generate";
+    private static final String VIEW_FORMATS = "formats";
+
+    private static final String ACTION_PARAM_ABSOLUTE_PATH = "absolutePath";
+    private static final String ACTION_PARAM_FILE_EXTENSION = "fileExtension";
+    private static final String ACTION_PARAM_SOURCE_DETAILS = "sourceDetails";
+    private static final String ACTION_PARAM_ALERT_SEVERITY = "alertSeverity";
+    private static final String ACTION_PARAM_ALERT_DETAILS = "alertDetails";
+
+    private ExtensionExportReport extension;
+
+    public ExportReportAPI(ExtensionExportReport extension) {
+        super();
+        this.extension = extension;
+
+        this.addApiAction(
+                new ApiAction( ACTION_GENERATE,
+                new String[] { ACTION_PARAM_ABSOLUTE_PATH,
+                               ACTION_PARAM_FILE_EXTENSION,
+                               ACTION_PARAM_SOURCE_DETAILS,
+                               ACTION_PARAM_ALERT_SEVERITY,
+                               ACTION_PARAM_ALERT_DETAILS }));
+        this.addApiView(new ApiView(VIEW_FORMATS));
+    }
+
+    @Override
+    public String getPrefix() {
+        return PREFIX;
+    }
+
+    @Override
+    public ApiResponse handleApiAction(String name, JSONObject params) throws ApiException {
+        if (logger.isDebugEnabled()) logger.debug("Request for handleApiAction: " + name + " (params: " + params.toString() + ")");
+
+        switch (name) {
+        case ACTION_GENERATE:
+            String absolutePath = params.getString(ACTION_PARAM_ABSOLUTE_PATH);
+
+            String fileExtension = params.getString(ACTION_PARAM_FILE_EXTENSION);
+
+            if (!extension.canWrite(absolutePath)) {
+                logger.error(Constant.messages.getString("exportreport.message.console.error.file.writable", absolutePath));
+                return ApiResponseElement.FAIL;
```

Ideally it should throw an exception (e.g.: `throw new ApiException(ApiException.Type.ILLEGAL_PARAMETER, ACTION_PARAM_ABSOLUTE_PATH);`) but since the Jenkins plugin expects this behaviour it should be left as is.
(When an exception is thrown core logs the issue.)